httpie/core.py

```diff
@@ -50,10 +50,8 @@ def main(args: List[Union[str, bytes]] = sys.argv, env=Environment()) -> ExitSta
     exit_status = ExitStatus.SUCCESS
 
     try:
-        parsed_args = parser.parse_args(
-            args=args,
-            env=env,
-        )
+        parsed_args = parser.parse_args(args=args, env=env)
+        exit_status = program(args=parsed_args, env=env)
```

Comment:
`parse_args()` and `program()` each have their own `try` clause intentionally because we handle different types of exceptions for them. For example, we don’t handle base `Exception`s from `parse_args()` as you can see. We also want to keep the `try` branches minimal and the distinction explicit. To get rid of the duplicate handling of `SystemError` and `KeyboardInterrupt`, something like this would be more in line with the intended behavior:

```python
try:
    parse_args()
    try:
        program()
    except Timeout:
        pass
    except TooManyRedirects:
        pass
    except Exception:
        pass
except KeyboardInterrupt:
    pass
except SystemExit:
    pass
```

Reply:
This approach would come with the implicit requirement that the outer `except` types don’t inherit from `Exception`, though.

Reply:
Good catch. I removed the commit as it is not really interesting to change that part for now.